File: back/app/entities/session.py
from app.constants.sessions import MAX_TITLE_LENGTH
from app.entities.application_entity import ApplicationEntity
from app.lib.mixin.validatable import model_name
from app.validations.blank_validation import BlankValidation
from app.validations.max_length_validation import MaxLengthValidation
from app.lib.mixin.validatable import validate
from app.lib.mixin.token_generatable import TokenGeneratable


@model_name('session')
@validate('title', BlankValidation)
@validate('title', MaxLengthValidation, MAX_TITLE_LENGTH)
@validate('token', BlankValidation)
@validate('created_at', BlankValidation)
class Session(ApplicationEntity, TokenGeneratable):
    def __init__(self, **attrs):
        super().__init__(**attrs)
        self.title = attrs.get('title')
        self.token = attrs.get('token')

File: front/src/models/session.ts
import moment from 'moment';
import { SessionResponse } from './interfaces/api/sessions';

export default class Session {
  token: string;
  title: string;
  createdAt: moment.Moment;
  updatedAt: moment.Moment;

  constructor({ token, title, created_at, updated_at }: SessionResponse) {
    this.token = token;
    this.title = title;
    this.createdAt = moment(created_at);
    this.updatedAt = moment(updated_at);
  }

  static get dummy(): Session {
    return new Session({ token: '', title: '', created_at: '', updated_at: '' });
  }
}
